crates/shared/src/oneinch_api.rs

```diff
@@ -517,36 +525,67 @@ impl OneInchClientImpl {
 
 #[async_trait::async_trait]
 impl OneInchClient for OneInchClientImpl {
-    async fn get_swap(&self, query: SwapQuery) -> Result<Swap, OneInchError> {
-        logged_query(&self.client, query.into_url(&self.base_url, self.chain_id)).await
+    async fn get_swap(
+        &self,
+        query: SwapQuery,
+        caching: Option<CachingStrategy>,
+    ) -> Result<Swap, OneInchError> {
+        logged_query(
+            &self.client,
+            query.into_url(&self.base_url, self.chain_id),
+            caching,
+        )
+        .await
     }
 
     async fn get_sell_order_quote(
         &self,
         query: SellOrderQuoteQuery,
+        caching: Option<CachingStrategy>,
     ) -> Result<SellOrderQuote, OneInchError> {
-        logged_query(&self.client, query.into_url(&self.base_url, self.chain_id)).await
+        logged_query(
+            &self.client,
+            query.into_url(&self.base_url, self.chain_id),
+            caching,
+        )
+        .await
     }
 
     async fn get_spender(&self) -> Result<Spender, OneInchError> {
         let endpoint = format!("v5.0/{}/approve/spender", self.chain_id);
         let url = crate::url::join(&self.base_url, &endpoint);
-        logged_query(&self.client, url).await
+        logged_query(
+            &self.client,
+            url,
+            Some(CachingStrategy::Time(Duration::from_secs(60))),
```

Comment:
I think adding code to do caching differently for these special enpoints is a premature optimisation. Can't this be done exclusively with configuration on the Nginx level (i.e. not specifying cache key headers for these specific routes?)

Reply:
I think those requests count to our quota so we definitely want to cache them (we have been doing this in the past if I'm not mistaken). Now, we can of course cache it in the nginx config specifically, but isn't it weird to have the option to specify caching behavior, choosing None and then still receive cached responses? One could argue that `None` should actually lead to populating a no-cache header.

Reply:
My point is in tandem to 👇  - in that I was arguing against using `Cache-Control` and using application specific headers and cache keys.